From cfccf244a76778132a59d79f802ff63a22e08f59 Mon Sep 17 00:00:00 2001
From: Ernesto Ramos <ernesto@ti.com>
Date: Wed, 28 Jul 2010 09:54:53 -0500
Subject: [PATCH 085/116] staging:ti dspbridge: proc_load/start should set IVA2 to OFF in case of failure

When a base image is being loaded or started and by some reason
the process fails, the IVA2 should be switched OFF.

Signed-off-by: Ernesto Ramos <ernesto@ti.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---
 drivers/staging/tidspbridge/rmgr/proc.c |    6 ++++--
 1 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/drivers/staging/tidspbridge/rmgr/proc.c b/drivers/staging/tidspbridge/rmgr/proc.c
index 6258d8b..44c26e1 100644
--- a/drivers/staging/tidspbridge/rmgr/proc.c
+++ b/drivers/staging/tidspbridge/rmgr/proc.c
@@ -1302,9 +1302,10 @@ int proc_load(void *hprocessor, const s32 argc_index,
 	}
 
 func_end:
-	if (status)
+	if (status) {
 		pr_err("%s: Processor failed to load\n", __func__);
-
+		proc_stop(p_proc_object);
+	}
 	DBC_ENSURE((!status
 		    && p_proc_object->proc_state == PROC_LOADED)
 		   || status);
@@ -1594,6 +1595,7 @@ func_cont:
 		}
 	} else {
 		pr_err("%s: Failed to start the dsp\n", __func__);
+		proc_stop(p_proc_object);
 	}
 
 func_end:
-- 
1.7.1

